tests/absolute_helpers/mod.rs

```diff
@@ -0,0 +1,197 @@
+use std::env;
+use std::fs;
+use std::io;
+use std::path;
+
+use path_abs::PathArc;
+
+use tempdir::TempDir;
+
+fn symlink_dir<P, Q>(src: P, dst: Q)
+where
+    P: AsRef<path::Path>,
+    Q: AsRef<path::Path>,
+{
+    #[cfg(windows)]
+    {
+        use std::os::windows::fs as winfs;
+
+        let dst = dst.as_ref();
+
+        winfs::symlink_dir(src, &dst).expect(
+            "Could not create symbolic link. \
+             Run as Administrator, or on Windows 10 in Developer Mode",
+        );
+        dst.symlink_metadata().expect(
+            "Link creation succeeded, but can't read link? \
+             If you're using Wine, see bug 44948",
+        );
+    }
+
+    #[cfg(unix)]
+    {
+        use std::os::unix::fs as unixfs;
+
+        unixfs::symlink(src, dst).expect("Could not create symbolic link");
+    }
+
+    #[cfg(all(not(windows), not(unix)))]
+    unreachable!();
+}
+
+#[test]
+fn absolute_path_is_idempotent() {
+    ::setup();
+    // The current_dir() result is always absolute,
+    // so absolutizing it should not change it.
+
+    let actual = PathArc::new(env::current_dir().unwrap())
+        .absolute()
+        .unwrap();
+    let expected = env::current_dir().unwrap().canonicalize().unwrap();
+
+    assert_eq!(actual.as_path(), expected.as_path());
+}
+
+#[test]
+fn absolute_path_removes_currentdir_component() {
+    ::setup();
+    let actual = PathArc::new("foo/./bar").absolute().unwrap();
+    let expected = PathArc::new("foo/bar").absolute().unwrap();
+
+    assert_eq!(actual, expected);
+}
+
+#[test]
+fn absolute_path_removes_empty_component() {
+    ::setup();
+    let actual = PathArc::new("foo//bar").absolute().unwrap();
+    let expected = PathArc::new("foo/bar").absolute().unwrap();
+
+    assert_eq!(actual, expected);
+}
+
+#[test]
+fn absolute_path_lexically_resolves_parentdir_component() {
+    ::setup();
+    let tmp_dir = TempDir::new("normalize_parentdir").unwrap();
+    let a_dir = tmp_dir.path().join("a");
+    fs::create_dir_all(&a_dir).unwrap();
+
+    let b_dir = tmp_dir.path().join("b");
+    fs::create_dir_all(&b_dir).unwrap();
+
+    fs::create_dir_all(&b_dir.join("target")).unwrap();
+
+    let link_path = a_dir.join("link");
+    symlink_dir("../b/target", link_path);
```

Comment:
I'm confused how this is `../` -- what is our CWD?

Reply:
The target of a (relative) symlink is always relative to the symlink itself; it doesn't matter what our CWD is.

```
$ cd $HOME
$ mkdir -p /tmp/somedir
$ ln -s ../otherdir /tmp/somedir/link
$ readlink -f /tmp/somedir/link
/tmp/otherdir
```

It's confusing, since it behaves differently to other two-argument operations like "copy" or "move", but that's the way it works.

Reply:
gotcha, makes sense.